Makefile

```diff
@@ -134,6 +134,7 @@ INC := -Iinclude -Iinclude/libc -Isrc -I$(BUILD_DIR) -I. -I$(EXTRACTED_DIR)
 
 # Check code syntax with host compiler
 CHECK_WARNINGS := -Wall -Wextra -Wno-format-security -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-variable -Wno-missing-braces
+CHECK_WARNINGS += -Werror=implicit-function-declaration
```

Comment:
What's this for?

Reply:
This is when using a function with no visible prototype
It's only a warning by default, should really just be an error
It's easy to forget to #include the relevant .h otherwise when moving prototypes from functions.h (whereas missing types/structs are errors)

Reply:
I can't read. I thought this was disabling this important warning. Either way, this feels like an odd time to introduce this even though I agree there's no situation where you'd want this to not error.

Reply:
Maybe consider splitting this into a separate PR? I agree for the rest of the changes but this could require more thought (maybe it has unintended consequences? And why is it the only one, what other warnings should be errors?)

Reply:
#1979